spec/models/az322_contribution_spec.rb

```diff
@@ -0,0 +1,124 @@
+#  updated_at              :datetime         not null
+#  state_file_az_intake_id :bigint
+#
+# Indexes
+#
+#  index_az322_contributions_on_state_file_az_intake_id  (state_file_az_intake_id)
+#
+require 'rails_helper'
+
+describe 'Az322Contribution' do
+  let(:intake) { create(:state_file_az_intake) }
+
+  describe "#made_contributions" do
+    it 'should validate presence' do
+      az = Az322Contribution.new(state_file_az_intake: intake, made_contribution: nil)
+      az.valid?
+      expect(az.errors[:made_contribution]).not_to be_empty
+
+      az.made_contribution = 'yes'
+      az.valid?
+      expect(az.errors[:made_contribution]).to be_empty
+
+      az.made_contribution = 'gobble'
+      az.valid?
+      expect(az.errors[:made_contribution]).not_to be_empty
+    end
+  end
+
+  context "made_contributions is true" do
+    let(:az) { Az322Contribution.new(state_file_az_intake: intake, made_contribution: 'yes') }
+
+    describe "#school_name" do
+      it 'should validate presence' do
+        az.school_name = nil
+        az.valid?
+        expect(az.errors[:school_name]).not_to be_empty
+
+        az.school_name = 'Hopscotch School'
+        az.valid?
+        expect(az.errors[:school_name]).to be_empty
+      end
+    end
+
+    describe "#ctds_code" do
+      it 'should validate presence' do
+        az.ctds_code = nil
+        az.valid?
+        expect(az.errors[:ctds_code]).not_to be_empty
+
+        az.ctds_code = '123456789'
+        az.valid?
+        expect(az.errors[:ctds_code]).to be_empty
+
+        az.ctds_code = '12345678'
+        az.valid?
+        expect(az.errors[:ctds_code]).not_to be_empty
+
+        az.ctds_code = 'fffffffff'
+        az.valid?
+        expect(az.errors[:ctds_code]).not_to be_empty
+      end
+    end
+
+    describe "#district_name" do
+      it 'should validate presence' do
+        az.district_name = nil
+        az.valid?
+        expect(az.errors[:district_name]).not_to be_empty
+
+        az.district_name = 'Sesame District'
+        az.valid?
+        expect(az.errors[:district_name]).to be_empty
+      end
+    end
+
+    describe "#amount" do
+      it 'should validate presence' do
+        az.amount = nil
+        az.valid?
+        expect(az.errors[:amount]).not_to be_empty
+
+        az.amount = 0
+        az.valid?
+        expect(az.errors[:amount]).not_to be_empty
+
+        az.amount = 1
+        az.valid?
+        expect(az.errors[:amount]).to be_empty
+      end
+
+      it "validates two or less decimal places" do
+        az.amount = 100.11
+        az.valid?
+        expect(az.errors[:amount]).to be_empty
+
+        az.amount = 100.3
+        az.valid?
+        expect(az.errors[:amount]).to be_empty
+
+        az.assign_attributes(amount: 100.113)
+        az.valid?
+        expect(az.errors[:amount]).not_to be_empty
+      end
+    end
+
+    describe "#date_of_contribution_is_valid_date" do
+      it 'should validate presence' do
+        # TODO
```

Comment:
@tahsinaislam I wasn't sure how to write this test, you probably have more context on what that validation was expected to do